srv/SetSimulationPaused.srv

```diff
@@ -0,0 +1,9 @@
+# Pauses or unpauses the simulation
```

Comment:
In addition to Pause, I'm missing a Step interface that would allow stepping the sim N steps at a time and then pausing.

Reply:
Could you elaborate on this? Would the service be blocking? What would the service look like, is there an implementation you would like to refer to? How would an example use look like?

Not sure if all the simulators support stepping by N.

Reply:
This was definitely possible in Gazebo Classic. And it is a super important thing for e.g. machine learning/RL.

https://github.com/gazebosim/gazebo-classic/blob/b22c6e15e52299865b31093b8feebc9ca19e26e8/gazebo/physics/World.cc#L1625

https://github.com/gazebosim/gazebo-classic/blob/b22c6e15e52299865b31093b8feebc9ca19e26e8/tools/gz.cc#L215

https://github.com/gazebosim/gazebo-classic/blob/b22c6e15e52299865b31093b8feebc9ca19e26e8/tools/gz_TEST.cc#L518

Reply:
Thinking about it.. Would you consider it makes sense for step simulation to be an action instead? Reporting feedback on each step, cancellable etc?

Reply:
Pushed out the service. I will give action a bit more thought - what would be the best standard feedback (except for step count).